packages/fxa-auth-server/test/local/senders/email.js

```diff
@@ -16,216 +16,810 @@ const sinon = require('sinon');
+  flowId: '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+  ip: '219.129.234.194',
+  locations: [],
+  numberRemaining: 2,
+  primaryEmail: 'c@d.com',
+  productId: 'wibble',
+  service: 'sync',
+  timeZone: 'America/Los_Angeles',
+  tokenCode: 'abc123',
+  type: 'secondary',
+  uaBrowser: 'Firefox',
+  uaBrowserVersion: '70.0a1',
+  uaOS: 'Windows',
+  uaOSVersion: '10',
+  uid: 'uid',
+  unblockCode: 'AS6334PK',
+};
+
+// key = query param name, value = MESSAGE property name
+const MESSAGE_PARAMS = new Map([
+  ['code', 'code'],
+  ['email', 'email'],
+  ['primary_email_verified', 'email'],
+  ['product_id', 'productId'],
+  ['secondary_email_verified', 'email'],
+  ['service', 'service'],
+  ['uid', 'uid'],
+  ['unblockCode', 'unblockCode'],
 ]);
 
-const typesContainPasswordManagerInfoLinks = new Set([
-  'passwordResetRequiredEmail',
+const COMMON_TESTS = new Map([
+  ['from', { test: 'equal', expected: config.smtp.sender }],
+  ['sender', { test: 'equal', expected: config.smtp.sender }],
+  [
+    'headers',
+    new Map([
+      ['X-Device-Id', { test: 'equal', expected: MESSAGE.deviceId }],
+      ['X-Email-Service', { test: 'equal', expected: 'fxa-auth-server' }],
+      ['X-Flow-Begin-Time', { test: 'equal', expected: MESSAGE.flowBeginTime }],
+      ['X-Flow-Id', { test: 'equal', expected: MESSAGE.flowId }],
+      ['X-Service-Id', { test: 'equal', expected: MESSAGE.service }],
+      [
+        'X-SES-CONFIGURATION-SET',
+        { test: 'equal', expected: config.smtp.sesConfigurationSet },
+      ],
+      ['X-Uid', { test: 'equal', expected: MESSAGE.uid }],
+    ]),
+  ],
+  [
+    'text',
+    [
+      // Ensure no HTML character entities appear in plaintext emails, &amp; etc
+      { test: 'notMatch', expected: /(?:&#x?[0-9a-f]+;)|(?:&[a-z]+;)/i },
+    ],
+  ],
 ]);
 
-const typesContainManageSettingsLinks = new Set([
-  'newDeviceLoginEmail',
-  'postVerifySecondaryEmail',
-  'postChangePrimaryEmail',
-  'postRemoveSecondaryEmail',
-  'postAddTwoStepAuthenticationEmail',
-  'postRemoveTwoStepAuthenticationEmail',
-  'postNewRecoveryCodesEmail',
-  'postConsumeRecoveryCodeEmail',
-  'postRemoveAccountRecoveryEmail',
-]);
-
-const typesContainRecoveryCodeLinks = new Set(['lowRecoveryCodesEmail']);
-
-const typesPrependVerificationSubdomain = new Set([
-  'verifyEmail',
-  'verifyLoginEmail',
+// prettier-ignore
+const TESTS = new Map([
+  ['downloadSubscriptionEmail', new Map([
+    ['subject', { test: 'equal', expected: 'Welcome to Secure Proxy!' }],
+    ['headers', new Map([
+      ['X-SES-MESSAGE-TAGS', { test: 'equal', expected: sesMessageTagsHeaderValue('downloadSubscription') }],
+      ['X-Template-Name', { test: 'equal', expected: 'downloadSubscription' }],
+      ['X-Template-Version', { test: 'equal', expected: SUBSCRIPTION_TEMPLATE_VERSIONS.downloadSubscription }],
+    ])],
+    ['html', [
+      { test: 'include', expected: configHref('privacyUrl', 'new-subscription', 'privacy') },
```

Comment:
And this is an example of the helper being invoked. If it obscures the guts of the assertion too much we can revert that commit, no worries.